model/src/ais_decoder.cpp

```diff
@@ -3197,6 +3197,11 @@ bool AisDecoder::Parse_VDXBitstring(AisBitstring *bstr,
                     sa.latitude = bstr->GetInt(base + 31, 24, true) / 60000.0;
                     break;
                   case AIS8_001_22_SHAPE_POLYLINE:
+                    for (int i = 0; i < 4; ++i) {
+                      sa.angles[i] = bstr->GetInt(base + 6 + i * 20, 10) * 0.5;
+                      sa.dists_m[i] =
+                          bstr->GetInt(base + 16 + i * 20, 10) * scale_factor;
+                    }
```

Comment:
This is not necessary, it just duplicates the code bellow and until now handled both the cases.

Reply:
Testing with VDR_pi did not draw a polyline. Hence the duplication.
[ais8_1_22_polyline.txt](https://github.com/OpenCPN/OpenCPN/files/14483528/ais8_1_22_polyline.txt)